packages/compiler-cli/src/ngtsc/typecheck/src/oob.ts

```diff
@@ -211,6 +223,50 @@ export class OutOfBandDiagnosticRecorderImpl implements OutOfBandDiagnosticRecor
         templateId, mapping, diagnosticVar.keySpan, ts.DiagnosticCategory.Suggestion,
         ngErrorCode(ErrorCode.SUGGEST_SUBOPTIMAL_TYPE_INFERENCE), message));
   }
+
+  splitTwoWayBinding(
+      templateId: TemplateId, input: BoundAttribute, output: TmplAstBoundEvent,
+      inputConsumer: ClassDeclaration, outputConsumer: ClassDeclaration|TmplAstElement): void {
+    const mapping = this.resolver.getSourceMapping(templateId);
+    const errorMsg = `The property and event halves of the two-way binding '${
+        input.name}' are not bound to the same target.
+            Find more at https://angular.io/guide/two-way-binding#how-two-way-binding-works`;
+
+    let relatedMessages = [];
+
+    relatedMessages.push({
+      text: `The property is bound to the '${inputConsumer.name.getText()}' component.`,
+      start: inputConsumer.name.getStart(),
+      end: inputConsumer.name.getEnd(),
+      sourceFile: inputConsumer.name.getSourceFile(),
+    });
+
+    if (outputConsumer instanceof TmplAstElement) {
+      let message = `The output binding is to a native event called '${input.name}' on the '${
+          outputConsumer.name}' DOM element.`;
+      if (!mapping.node.getSourceFile().isDeclarationFile) {
+        message += `\n \n Are you missing an output declaration called '${output.name}'?`;
+      }
+      relatedMessages.push({
+        text: message,
+        start: outputConsumer.sourceSpan.start.offset,
+        end: outputConsumer.sourceSpan.end.offset,
+        sourceFile: mapping.node.getSourceFile(),
+      });
+    } else {
+      relatedMessages.push({
+        text: `The event is bound to the '${outputConsumer.name.getText()}' component.`,
+        start: outputConsumer.name.getStart(),
+        end: outputConsumer.name.getEnd(),
+        sourceFile: outputConsumer.name.getSourceFile(),
+      });
+    }
+
+
+    this._diagnostics.push(makeTemplateDiagnostic(
+        templateId, mapping, input.sourceSpan, ts.DiagnosticCategory.Error,
```

Comment:
```suggestion
        templateId, mapping, input.keySpan, ts.DiagnosticCategory.Error,
```

This will limit the diagnostic to highlight the left side of the binding (`[(foo)]`) instead of the whole expression, which looks cleaner.